src/core/or/connection_or.c

```diff
@@ -1364,10 +1366,34 @@ void
 connection_or_connect_failed(or_connection_t *conn,
                              int reason, const char *msg)
 {
+  const or_options_t *options = get_options();
+
   control_event_or_conn_status(conn, OR_CONN_EVENT_FAILED, reason);
-  if (!authdir_mode_tests_reachability(get_options()))
+  if (!authdir_mode_tests_reachability(options))
     control_event_bootstrap_prob_or(msg, reason, conn);
   note_or_connect_failed(conn);
+
+  if (options->ClientAutoIPv6ORPort) {
+    or_state_t *state = get_or_state();
+    switch (tor_addr_family(&conn->real_addr)) {
+      case AF_INET:
+        if (reason == END_OR_CONN_REASON_NO_ROUTE) {
```

Comment:
We don't want to turn off an IP family just because our routing table doesn't have an entry for one IP address. Maybe we should track the number of failed connections due to no route to host.